File: src/code/python/binary_search/greedy_maximum.py
def fn(arr):
    def check(x):
        return BOOLEAN

    left = MINIMUM_POSSIBLE_ANSWER
    right = MAXIMUM_POSSIBLE_ANSWER

    while left <= right:
        mid = (left + right) // 2

        if check(mid):
            left = mid + 1
        else:
            right = mid - 1

    return right

File: src/code/ruby/binary_search/greedy_maximum.rb
def fn(arr)
  left  = MINIMUM_POSSIBLE_ANSWER
  right = MAXIMUM_POSSIBLE_ANSWER

  while left <= right
    mid = (left + right) / 2 # integer division

    if check(mid)
      left = mid + 1
    else
      right = mid - 1
    end

  end

  right
end

def check(x)
  BOOLEAN
end
